apps/server/src/clerk/clerk.guard.ts

```diff
@@ -0,0 +1,30 @@
+import { ExecutionContext, Injectable } from '@nestjs/common'
+import { AuthGuard } from '@nestjs/passport'
+import { ClerkService } from './clerk.service'
+
+@Injectable()
+export class ClerkAuthGuard extends AuthGuard('jwt') {
+  constructor(private readonly clerkService: ClerkService) {
+    super()
+  }
+
+  async canActivate(context: ExecutionContext) {
+    const request = context.switchToHttp().getRequest()
+    const authToken = request.headers.authorization
+      ? request.headers.authorization.replace('Bearer ', '')
+      : request.headers.authorization
+
+    if (!authToken) {
+      return false
+    }
+
+    const user = await this.clerkService.authenticateUser(authToken)
+
+    if (!user) {
+      return false
+    }
+
+    request.user = user
+    return true
+  }
```

Comment:
```suggestion
    return Boolean(user)
  }
```

Curiosidade, isso resolveria?

Reply:
resolveria para o return true ou false, mas não para o usuário ser adicionado ao request object, acredito também que da primeira forma fique mais simples de entender, lendo por cima do código..